src/packet.rs

```diff
@@ -156,5 +210,35 @@ packet_boilerplate!(
     (1, StatusRequest, 0, []),
     (1, Ping, 1, [(payload, Long)]),
     (11, Pong, 1, [(payload, Long)]),
-    (11, StatusResponse, 0, [(json_response, String)])
+    (11, StatusResponse, 0, [(json_response, String)]),
+    (2, LoginStart, 0, [(username, String)]),
+    (2, LoginSuccess, 2, [(uuid, String), (username, String)]),
+    (
+        3,
+        JoinGame,
+        25,

```

Comment:
Nope- the id of join game listed in the protocol is *0x25* which is a hex number. That's why your join packet didn't work and also why your next packet didn't work